python/plugins/processing/gui/wrappers.py

```diff
@@ -1615,14 +1620,26 @@ def postInitialize(self, wrappers):
                 break
 
     def parentValueChanged(self, wrapper):
-        self.setLayer(wrapper.parameterValue())
+        value = wrapper.parameterValue()
+        if value in wrapper.fileBasedLayers:
```

Comment:
What's this code for?

Reply:
If the loaded layer is in the cache, it sets the widget layer with it instead of the parameter value. Otherwise, it just calls set layer with value itself.

Reply:
It's not directly required by this PR though, is it? The `QgsProcessingUtils.mapLayerFromString` call used in setLayer already effectively uses a cache inside self.context, so we end up double-caching here, at a cost of more code complexity.

Reply:
Not really required, but the performance improvement is noticeable. Otherwise, for an algorithm such as "Create point layer from table", with 4 dependent fields, mapLayerFromString is called 4 times, and the warning message is shown with a rather long delay.

I didnt know there was a cache in that method, but looks like it is not working in this case. Maybe mapLayerFromString is not caching when the layer cannot be read (wrong layer file, as in this case), and for this reason it tries 4 times to load the wrong layer (all of them unsuccessfully)

Reply:
> Maybe mapLayerFromString is not caching when the layer cannot be read

Ah, you're right. All good here :)